### ISSUE TITLE
Support exponential stop functions

### ISSUE THREAD

**Issue by ianthetechie** [stadiamaps/cartogrify#5]
Mapbox GL supports stops (for width, colors, etc.) that follow both a linear and exponential progression. Tangram does not appear to support this directly, but cartogrify could reasonably fake it by computing the intermediate stop values.

Current status:

- [x] Support numeric interpolation (https://github.com/mapbox/mapbox-gl-js/blob/f0cc0159819617f59bbb5c3405a13a22f2f5c704/src/style-spec/util/interpolate.js#L5-L7)
- [ ] Support color interpolation (https://github.com/mapbox/mapbox-gl-js/blob/f0cc0159819617f59bbb5c3405a13a22f2f5c704/src/style-spec/util/interpolate.js#L9-L16}
- [ ] Support array interpolation (https://github.com/mapbox/mapbox-gl-js/blob/f0cc0159819617f59bbb5c3405a13a22f2f5c704/src/style-spec/util/interpolate.js#L18-L22)

**Comment by bcamper** [comment]
For now, you would need to implement this as a custom JS function, such as one based on Mapbox GL's (last OS version v1.13.0):

https://github.com/mapbox/mapbox-gl-js/blob/f87090b70b8a6fbde1a779c44933ce32bffc5d2c/src/style-spec/expression/definitions/interpolate.js#L254-L265

I am interested in adding direct support for exponential interpolation in Tangram though, it's long been on the wishlist, just never got around to it.

**Comment by ianthetechie** [comment]
Yeah, I assumed as much. I've implemented a few similar things with JS functions already. Thanks a bunch for the direct source link! That will exponentially speed up my implementation of this ;)
